src/validation.cpp

```diff
@@ -5337,63 +5337,3 @@ class CMainCleanup
         mapBlockIndex.clear();
     }
 } instance_of_cmaincleanup;
-
-// ELEMENTS:
-/* This function checks that the RPC connection to the parent chain node
- * can be attained, and is returning back reasonable answers.
- */
-bool MainchainRPCCheck(const bool init)
-{
-    // Check for working and valid rpc
-    if (gArgs.GetBoolArg("-validatepegin", Params().GetConsensus().has_parent_chain)) {
```

Comment:
Because it's only ever called inside a conditional that already checks the same condition, so it was redundant.

(It sounded like you had an example of this failing though -- can you provide details?)